honeybee_3dm/togeometry.py

```diff
@@ -1,34 +1,311 @@
+
+    Returns:
+        A Ladybug Mesh3D object
+    """
+    lb_verts = tuple(to_point3d(mesh.Vertices[i]) for i in range(len(mesh.Vertices)))
+    lb_faces, colors = extract_mesh_faces_colors(mesh, color_by_face)
+    return Mesh3D(lb_verts, lb_faces, colors)
+
+
+def brep_to_face3d(brep):
+    """Create a Ladybug Face3D object from a rhino3dm Brep.
+
+    Args:
+        brep: A rhino3dm Brep.
+
+    Returns:
+        A list of Ladybug Face3D objects.
+    """
+    faces = []
+    for i in range(len(brep.Faces)):
+        mesh = brep.Faces[i].GetMesh(rhino3dm.MeshType.Any)
+        faces.extend(mesh_to_face3d(mesh))
+
+    return faces
+
+
+def brep2d_to_face3d(brep, tolerance):
+    """Create a Ladybug Face3D object from a rhino3dm Brep.
+
+    This function is used in translating rhino3dm planar Brep to Ladybug Face3D objects.
+
+    Args:
+        brep: A rhino3dm Brep.
+
+    Returns:
+        A list of Ladybug Face3D objects.
+    """
+    faces = []
+    # * Check 01 - If any of the edge is an arc
+    lines = []
+    not_line = 0
+    for i in range(len(brep.Edges)):
+        if brep.Edges[i].IsLinear(tolerance) is False:
+            not_line += 1
+    # If there's an arc in one of the edges, mesh it
+    if not_line > 0:
+        warnings.warn(
+            "The surface has curved edges. It will be meshed.")
+        faces.extend(brep_to_face3d(brep))
+        return faces
+    else:
+        # Create Ladybug lines from start and end points of edges
+        for i in range(len(brep.Edges)):
+            start_pt = to_point3d(brep.Edges[i].PointAtStart)
+            end_pt = to_point3d(brep.Edges[i].PointAtEnd)
+            line = LineSegment3D.from_end_points(start_pt, end_pt)
+            lines.append(line)
+
+        # Create Ladybug Polylines from the lines
+        polylines = Polyline3D.join_segments(lines, tolerance)
+        # Sort the Ladybug Polylines based on length
+        # The longest polyline belongs to the boundary of the face
+        # The rest of the polylines belong to the holes in the face
+        sorted_polylines = sorted(polylines, key=lambda polyline: polyline.length)
+        # The first polyline in the list shall always be the polyline for
+        # the boundary
+        sorted_polylines.reverse()
+
+        # Getting all vertices of the face
+        mesh = brep.Faces[0].GetMesh(rhino3dm.MeshType.Any)
+        try:
+            mesh_pts = [to_point3d(mesh.Vertices[i])
+                        for i in range(len(mesh.Vertices))]
+        except AttributeError:
+            warnings.warn(
+                'Non-planar Breps are not handled by Honeybee. They are ignored.'
+            )
+            return faces
+        else:
+            # In the list of the Polylines, if there's only one polyline then
+            # the face has no holes
+            if len(sorted_polylines) == 1:
+                if len(mesh_pts) == 4 or len(mesh_pts) == 3:
```

Comment:
Couldn't you just do this check at the very start?  If there are only 3 or 4 vertices, then there's no need to do this whole crazy thing with the polyline edges, etc.

Reply:
Or maybe you could check if the number of Brep edges is 3 or 4.   Also, a hole is impossible for any shape with 5 edges or less so maybe you can save yourself a lot of extra checks in these cases by just ignoring the search for holes in these cases.

Reply:
I have revised as per the comment above. I don't look for holes once there's only one Polyline.